adal/src/main/java/com/microsoft/aad/adal/AcquireTokenSilentHandler.java

```diff
@@ -128,7 +129,32 @@ AuthenticationResult acquireTokenWithRefreshToken(final String refreshToken)
                 Logger.i(TAG, "Refresh token is not returned or empty", "");
                 result.setRefreshToken(refreshToken);
             }
-        } catch (final IOException | AuthenticationException exc) {
+        } catch (final AuthenticationException exc) {
+            //if get 503,504,500 && outageMode is on, return the stale token
+            if (exc.getCode().equals(ADALError.SERVER_NOT_RESPONDING) && mOutageModeIsOn) {
+                final TokenCacheItem accessTokenItem = mTokenCacheAccessor.getRegularRefreshTokenCacheItem(mAuthRequest.getResource(),
+                        mAuthRequest.getClientId(), mAuthRequest.getUserFromRequest());
+                if (accessTokenItem.getAccessToken() != null
+                        && accessTokenItem.getExtendedExpiresOn() != null
```

Comment:
>accessTokenItem.getExtendedExpiresOn() != null [](start = 27, length = 46)

This function can never return null, look at the implementation in TokenCacheItem.java